openedx/core/djangoapps/import_export/courses/views.py

```diff
@@ -0,0 +1,528 @@
+"""
+These views handle all actions in Studio related to import and exporting of
+courses
+"""
+import base64
+import logging
+from opaque_keys import InvalidKeyError
+import os
+import re
+import shutil
+import tarfile
+from path import path  # pylint: disable=no-name-in-module
+
+from django.conf import settings
+from django.core.cache import cache
+from django.core.exceptions import SuspiciousOperation
+from django.core.files.temp import NamedTemporaryFile
+from django.core.servers.basehttp import FileWrapper
+from django.http import HttpResponse, Http404
+from django.utils.translation import ugettext as _
+from django.shortcuts import redirect
+
+from rest_framework import renderers
+from rest_framework.authentication import OAuth2Authentication, \
+    SessionAuthentication
+from rest_framework.decorators import renderer_classes \
+    as renderer_classes_decorator
+from rest_framework.permissions import IsAuthenticated, BasePermission
+from rest_framework.renderers import JSONRenderer
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+import dogstats_wrapper as dog_stats_api
+from xmodule.contentstore.django import contentstore
+from xmodule.exceptions import SerializationError
+from xmodule.modulestore.django import modulestore
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys.edx.locator import LibraryLocator
+from xmodule.modulestore.xml_importer import import_course_from_xml, import_library_from_xml
+from xmodule.modulestore.xml_exporter import export_course_to_xml, export_library_to_xml
+from xmodule.modulestore import COURSE_ROOT, LIBRARY_ROOT
+
+from student.auth import has_course_author_access
+
+from openedx.core.lib.extract_tar import safetar_extractall
+from openedx.core.lib.tempdir import mkdtemp_clean
+from util.json_request import JsonResponse
+from util.views import ensure_valid_course_key
+
+from urllib import urlencode
+
+log = logging.getLogger(__name__)
+
+
+# Regex to capture Content-Range header ranges.
+CONTENT_RE = re.compile(
+    r"(?P<start>\d{1,11})-(?P<stop>\d{1,11})/(?P<end>\d{1,11})"
+)
+
+
+class IsCourseStaff(BasePermission):
+    """
+    Permission that checks to see if the request user has permission to access
+    all course content of the requested course
+    """
+    def has_permission(self, request, view):
+        course_key_string = view.kwargs['course_key_string']
+        try:
+            course_key = CourseKey.from_string(course_key_string)
+        except InvalidKeyError:
+            raise Http404
+
+        return has_course_author_access(request.user, course_key)
```

Comment:
This appears to be a deprecated function (it just redirects to `has_studio_write_access`). That in itself is somewhat misleading, since we're making this not a studio-only thing anymore. @ormsbee, @nasthagiri: This appears to be a prime candidate for any upcoming roles/permissions improvements.